src/ufoLib2/objects/component.py

```diff
@@ -1,14 +1,19 @@
 import warnings
-from typing import Optional, Tuple
+from typing import TYPE_CHECKING, Optional, Tuple
 
 import attr
 from fontTools.misc.transform import Identity, Transform
-from fontTools.pens.pointPen import PointToSegmentPen
+from fontTools.pens.basePen import AbstractPen
+from fontTools.pens.pointPen import AbstractPointPen, PointToSegmentPen
 
+from ufoLib2.objects.misc import BoundingBox
 from ufoLib2.typing import Number
 
 from .misc import _convert_transform, getBounds, getControlBounds
 
+if TYPE_CHECKING:
+    from ufoLib2.objects.layer import Layer
```

Comment:
Out of curiosity, why does this have to be `if TYPE_CHECKING`?

Reply:
To avoid a circular import at runtime but make mypy type check the code.